polaris/hub/client.py

```diff
@@ -662,18 +720,193 @@ def upload_benchmark(
+        """List all available competitions on the Polaris Hub.
+
+        Args:
+            limit: The maximum number of competitions to return.
+            offset: The offset from which to start returning competitions.
+
+        Returns:
+            A list of competition names in the format `owner/competition_name`.
+        """
+
+        # TODO (cwognum): What to do with pagination, i.e. limit and offset?
+        response = self._base_request_to_hub(
+            url="/v2/competition", method="GET", params={"limit": limit, "offset": offset}
+        )
+        benchmarks_list = [f"{HubOwner(**bm['owner'])}/{bm['name']}" for bm in response["data"]]
+        return benchmarks_list
+
+    def evaluate_competition(
+        self,
+        competition: CompetitionSpecification,
+        y_pred: PredictionsType,
+        result_name: str,
+        description: str = "",
+        tags: list[str] = [],
+        user_attributes: Dict[str, str] = {},
+        github_url: HttpUrlString | None = "",
+        paper_url: HttpUrlString | None = "",
+        contributors: list[HubUser] | None = [],
+        results_access: AccessType = "private",
+    ) -> CompetitionResults:
+        """Evaluate the predictions for a competition on the Polaris Hub. Target labels are fetched
+        by Polaris Hub and used only internally.
+
+        Args:
+            competition: The competition to evaluate the predictions for.
+            y_pred: The predictions for the test set, as NumPy arrays.
+                If there are multiple targets, the predictions should be wrapped in a dictionary with the target labels as keys.
+
+        Returns:
+             A `CompetitionResults` object.
+        """
+
+        y_pred = serialize_predictions(y_pred)
+        response = self._base_request_to_hub(
+            url=f"/v2/competition/{competition.owner}/{competition.name}/evaluate",
+            method="POST",
+            json={
+                "predictions": y_pred,
+                "access": results_access,
+                "polaris_version": po.__version__,
+                "name": result_name,
+                "description": description,
+                "tags": tags,
+                "user_attributes": user_attributes,
+                "github_url": github_url,
+                "paper_url": paper_url,
+                "contributors": contributors,
+            },
+        )
+
+        # Inform the user about where to find their newly created artifact.
+        result_url = urljoin(
+            self.settings.hub_url,
+            f"/v2/competition/{competition.owner}/{competition.name}/{response['id']}",
+        )
+        logger.success(f"Your result has been successfully uploaded to the Hub. View it here: {result_url}")
+
+        scores = response["results"]
+        return CompetitionResults(
+            results=scores, competition_name=competition.name, competition_owner=competition.owner
+        )
```

Comment:
**Fix:** All the additional metadata that is part of the method signature should also be included in the object that is returned here I believe.